pennylane/_device.py

```diff
@@ -678,9 +679,9 @@ def default_expand_fn(self, circuit, max_expansion=10):
         )
         obs_on_same_wire = len(circuit._obs_sharing_wires) > 0 or comp_basis_sampled_multi_measure
         obs_on_same_wire &= not any(
-            isinstance(o, (Hamiltonian, LinearCombination)) for o in circuit._obs_sharing_wires
+            isinstance(o, (Hamiltonian, LinearCombination, Prod))
```

Comment:
do we understand why `Prod` needs to be included here but `Tensor` doesn't for the test to pass?

Reply:
This is most likely because, with opmath disabled, we have a `Hamiltonian` instance rather than a `Tensor.` 

```
type(2 * qml.PauliX(0) @ qml.PauliY(1))
```
results in:
```
pennylane.ops.qubit.hamiltonian.Hamiltonian
```

The expression cannot be a `Tensor` because it has a scalar factor. The test also fails if we run it with opmath disabled and remove the `2 *` from the expression (in that case the expression becomes a `Tensor`), unless we add `qml.operation.Tensor` in the instance check :)

Reply:
I'm hesitant to meddle with `_device.py`.

If the problem is that now with new opmath enabled, `2 * qml.PauliX(0) @ qml.PauliY(1)` becomes a `Prod` while it should be a `Hamiltonian` (/`LinearCombination`) then I would suggest to change the test to use `qml.Hamiltonian([2], [qml.X(0) @ qml.Y(1)])` instead. Then it _should_ pass for both.